lib/pry/input_completer.rb

```diff
@@ -239,4 +227,39 @@ def build_path(input)
     end
     return path, input
   end
+
+  private
+
+  def self.fetch_obj_space_collection
```

Comment:
And if you already have `class << self` section on the top, it would be better to move this methods there.

Mutext can be also set once with a constant: `MUTEXT = Mutex.new`